### PR TITLE
Add new rule `no-string-prototype-extensions`

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves https://github.com/ember-cli/eslint-plugin-ember/issues/638

### CODE REVIEW

File: tests/lib/rules/no-string-prototype-extensions.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+});
+
+ruleTester.run('no-string-prototype-extensions', rule, {
+  valid: [
+    "dasherize('myString')",
+    'capitalize(someString)',
+    "htmlSafe('<b>foo</b>')",
+    "someString['foo']()",
+    'something.foo()',
+  ],
```
Comment (bmish) [inline tests/lib/rules/no-string-prototype-extensions.js:20]: Can you add `dasherize.foo()` as a valid example?

Comment (bmish) [inline tests/lib/rules/no-string-prototype-extensions.js:20]: How about `getSomeString().dasherize()`?

Comment (bmish) [inline tests/lib/rules/no-string-prototype-extensions.js:20]: `this.dasherize()` is valid right.

Comment (Turbo87) [reply tests/lib/rules/no-string-prototype-extensions.js:22]: done

Comment (Turbo87) [reply tests/lib/rules/no-string-prototype-extensions.js:22]: added

Comment (Turbo87) [reply tests/lib/rules/no-string-prototype-extensions.js:22]: added and implementation adjusted :)

**Review by bmish — APPROVED** [review]

**Pull request merged**